### ISSUE TITLE
Cloudwatch assume role not working after 5.4.3

### ISSUE THREAD

**Issue by boostmikal** [grafana/grafana#16725]
**What happened**:
Using Cloudwatch datasource with arn authentication does not work.

The Cloudwatch datasource config used:
 * **Auth Provider**: ARN
 * **Assume Role ARN**: _empty_
 * **Default Region**: eu-west-1
 * **Custom Metrics**: _empty_

This config gives error in the GUI:
`InvalidParameter: 1 validation error(s) found.
minimum field size of 20, AssumeRoleInput.RoleArn.`

And the following error in the log.
`t=2019-04-23T13:15:22+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=*****@*****.com error="Failed to call ec2:getAwsConfig, AccessDenied: Access denied\n\tstatus code: 403, request id: d9d9c45c-65c9-11e9-8c23-f595b58c78b3"`

**What you expected to happen**:
Being able to fetch metrics from Cloudwatch.

**How to reproduce it (as minimally and precisely as possible)**:
* Configure Cloudwatch in Grafana 6.1.4 without specifying Assume Role ARN.
* Try to make a graph using Cloudwatch metrics or try to use Explore with the Cloudwatch datasource

**Anything else we need to know?**:
I am running this in Docker with the images provided by Grafana on DockerHub.

The setup described above is working on as expected in **5.4.3** runnning in Docker by only giving the Auth Provider and Default Region in the Cloudwatch datasource config. I have tested with **6.1.4** and **6.0.0-beta1**, to me it looks like the Cloudwatch datasource with automatic ARN discovery and role assumption on AWS stopped working after 5.4.3.

**Environment**:
- Grafana version: 6.1.4
- Data source type & version: Cloudwatch
- OS Grafana is installed on: Ubuntu 16.04, running in Docker
- User OS & Browser: Ubuntu 16.04, Chrome
- Grafana plugins: none, only bundled plugins
- Others: Docker CE 18.09

**Comment by marefr** [comment]
Question is then if Grafana should enable shared configuration file per default or not?

**Comment by marefr** [comment]
@mtanda do you any thoughts?

**Comment by mtanda** [comment]
`aws-sdk-go` add validation for the input.
https://github.com/aws/aws-sdk-go/commit/e8e504c4dda008a5f9ed95e751f00cdd9bc21587#diff-7466ca3a7215b63e125978ef806c7174R508

I guess the datasource setting is incorrect, it should be like `arn:aws:iam::123456789012:role/foobar`.

**Comment by boostmikal** [comment]
@mtanda You are correct, it makes sense that the ARN auth provider requires an actual arn to work, even though it did not prior to 6.0.0. I did get the ARN auth provider to work on both 5.4.3 and 6.1.4 using the assumed role ARN connected to the instance profile of the EC2 instance.

What I wanted Grafana to do was to automatically assume the role identified by the `get-caller-id` of the AWS API. It looks like this is possible by using the Credentials File auth provider. I found this in a Hacker Noon article - https://hackernoon.com/aws-cloudwatch-monitoring-with-grafana-ace63e1ab507

Thank you for your help, from my side I consider the issue fixed.

For reference to anyone else who might need it, I was provisioning the Cloudwatch datasource like this before.

```
apiVersion: 1

datasources:
  - name: CloudWatch
    type: cloudwatch
    jsonData:
      authType: arn
      defaultRegion: eu-west-1
```

And now I am provisioning like it like this. Same as found in the Grafana docs https://grafana.com/docs/features/datasources/cloudwatch/#configure-the-datasource-with-provisioning
```
apiVersion: 1

datasources:
  - name: CloudWatch
    type: cloudwatch
    jsonData:
      authType: credentials
      defaultRegion: eu-west-1
```

**Comment by mtanda** [comment]
Sometimes I confuse the configuration, I set authType to credentials too.
When I have a time, I'll check the code, and try to improve.

**Comment by marefr** [comment]
@mtanda I'm always confused looking at AWS IAM :smile: Sounds good :+1:

**Comment by kevinashaw** [comment]
I seem to be running into the same issue.  We are experimenting with a Grafana AMI running on an AWS EC2 instance.  I created a new role with the CloudWatchReadOnlyAccess policy and a trust relationship to "ec2.amazonaws.com".  I have bound that role to the EC2 instance.  I then created a new CloudWatch Data Source in Grafana, set it for ARN as the Auth Provider and entered the ARN of the new role.  Attempting to save gives me: "HTTP Error Internal Server Error".
I find the following error in the grafana.log file: 
`AccessDenied: User: arn:aws:sts::XXXXX:assumed-role/GrafanaRole/instanceid is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::XXXXX:role/GrafanaRole\n\tstatus code: 403`
Am I missing something?

**Comment by aknuds1** [comment]
@kevinashaw To me it sounds as if you shouldn't configure the data source to assume the role, since the EC2 instance is already bound to it. It looks from the error as the EC2 instance has assumed the role and tries to assume it again, since the data source was configured to do so. Make sense?

**Comment by kevinashaw** [comment]
I agree completely.  But Grafana demands that I enter an ARN for the Role.  I can't not provide it.

**Comment by aknuds1** [comment]
@kevinashaw If you're on v7.3.x try "AWS SDK Default" as the auth provider, otherwise try "Credentials file". There shouldn't be any need for you to pick ARN as the provider.

**Comment by kevinashaw** [comment]
I will try.  Thanks!
